pkg/bpfassets/attacher/libbpf_attacher.go

```diff
@@ -162,44 +162,14 @@ func libbpfCollectProcess() (processesData []ProcessBPFMetrics, err error) {
 	if err != nil {
 		return
 	}
-	//processKeySize := int(unsafe.Sizeof(uint64Key))
-	iterator := processes.Iterator()
-	var ct ProcessBPFMetrics
-	//valueSize := int(unsafe.Sizeof(ProcessBPFMetrics{}))
-	keys := []uint32{}
-	retry := 0
-	next := iterator.Next()
-	for next {
-		keyBytes := iterator.Key()
-		key := ByteOrder.Uint32(keyBytes)
-		data, getErr := processes.GetValue(unsafe.Pointer(&key))
-		if getErr != nil {
-			retry += 1
-			if retry > maxRetry {
-				klog.V(5).Infof("failed to get data: %v with max retry: %d \n", getErr, maxRetry)
-				next = iterator.Next()
-				retry = 0
-			}
-			continue
-		}
-		getErr = binary.Read(bytes.NewBuffer(data), ByteOrder, &ct)
-		if getErr != nil {
-			klog.V(5).Infof("failed to decode received data: %v\n", getErr)
-			next = iterator.Next()
-			retry = 0
-			continue
-		}
-		if retry > 0 {
-			klog.V(5).Infof("successfully get data with retry=%d \n", retry)
-		}
-		processesData = append(processesData, ct)
-		keys = append(keys, key)
-		next = iterator.Next()
-		retry = 0
+	if ebpfBatchGetAndDelete {
+		processesData, err = libbpfCollectProcessBatch(processes)
 	}
-	for _, key := range keys {
-		// TODO delete keys in batch
-		processes.DeleteKey(unsafe.Pointer(&key))
+	if err == nil {
+		return
+	} else {
+		ebpfBatchGetAndDelete = false
```

Comment:
I think we need log here to indicate it's not able to (so something might be wrong) ?

Reply:
> I think we need log here to indicate it's not able to (so something might be wrong) ?

Agree. We might log a message of an error and say that we are now changing to use single collect.